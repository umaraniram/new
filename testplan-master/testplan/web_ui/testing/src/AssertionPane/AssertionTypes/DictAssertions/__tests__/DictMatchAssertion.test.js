import React from 'react';
import {shallow} from 'enzyme';
import {StyleSheetTestUtils} from "aphrodite";

import DictMatchAssertion from '../DictMatchAssertion';

function defaultProps() {
  return {
    assertion: {
      "category":"DEFAULT",
      "machine_time":"2019-02-12T17:41:43.295236+00:00",
      "description":"Simple dict match",
      "comparison":[
        [0,"foo","Passed",["int","1"],["int","1"]],
        [0,"bar","Failed",["int","2"],["int","5"]],
        [0,"extra-key","Failed",[null,"ABSENT"],["int","10"]]
      ],
      "line_no":524,
      "expected_description":null,
      "actual_description":null,
      "meta_type":"assertion",
      "include_keys":null,
      "passed":false,
      "exclude_keys":null,
      "type":"DictMatch",
      "utc_time":"2019-02-12T17:41:43.295231+00:00"
    }
  };
}


describe('DictMatchAssertion', () => {
  let props;
  let shallowComponent;

  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
    props = defaultProps();
    shallowComponent = undefined;
  });

  it('shallow renders the correct HTML structure', () => {
    shallowComponent = shallow(<DictMatchAssertion {...props}/>);
    expect(shallowComponent).toMatchSnapshot();
  });
});