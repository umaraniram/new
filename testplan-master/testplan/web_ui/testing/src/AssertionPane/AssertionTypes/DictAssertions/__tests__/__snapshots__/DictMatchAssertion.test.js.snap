// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DictMatchAssertion shallow renders the correct HTML structure 1`] = `
<DictBaseAssertion
  buttons={
    <DictButtonGroup
      defaultSortType={3}
      flattenedDict={
        Array [
          Array [
            0,
            "bar",
            "Failed",
            Array [
              "int",
              "2",
            ],
            Array [
              "int",
              "5",
            ],
          ],
          Array [
            0,
            "extra-key",
            "Failed",
            Array [
              null,
              "ABSENT",
            ],
            Array [
              "int",
              "10",
            ],
          ],
          Array [
            0,
            "foo",
            "Passed",
            Array [
              "int",
              "1",
            ],
            Array [
              "int",
              "1",
            ],
          ],
        ]
      }
      setRowData={[Function]}
      sortTypeList={
        Array [
          1,
          2,
          3,
          4,
        ]
      }
    />
  }
  columns={
    Array [
      Object {
        "cellStyle": Object {
          "border": "1px solid",
          "borderColor": "#d9dcde",
          "fontSize": "small",
          "padding": "6px 11px 6px 11px",
        },
        "field": "key",
        "headerStyle": Object {
          "border": "1px solid",
          "borderColor": "#d9dcde",
        },
        "render": [Function],
        "title": "Key",
      },
      Object {
        "cellStyle": Object {
          "backgroundColor": "#BDC3C750",
          "border": "1px solid",
          "borderColor": "#d9dcde",
          "fontSize": "small",
          "padding": "6px 11px 6px 11px",
        },
        "field": "expected",
        "headerStyle": Object {
          "border": "1px solid",
          "borderColor": "#d9dcde",
        },
        "render": [Function],
        "title": "Expected",
      },
      Object {
        "cellStyle": Object {
          "backgroundColor": "#BDC3C750",
          "border": "1px solid",
          "borderColor": "#d9dcde",
          "fontSize": "small",
          "padding": "6px 11px 6px 11px",
        },
        "field": "value",
        "headerStyle": Object {
          "border": "1px solid",
          "borderColor": "#d9dcde",
        },
        "render": [Function],
        "title": "Value",
      },
    ]
  }
  rows={
    Array [
      Object {
        "descriptor": Object {
          "indent": 0,
          "isEmptyLine": false,
          "isListKey": false,
          "lineNo": 524,
          "status": "Failed",
        },
        "expected": Object {
          "type": "int",
          "value": "5",
        },
        "key": Object {
          "type": "key",
          "value": "bar",
        },
        "value": Object {
          "type": "int",
          "value": "2",
        },
      },
      Object {
        "descriptor": Object {
          "indent": 0,
          "isEmptyLine": false,
          "isListKey": false,
          "lineNo": 524,
          "status": "Failed",
        },
        "expected": Object {
          "type": "int",
          "value": "10",
        },
        "key": Object {
          "type": "key",
          "value": "extra-key",
        },
        "value": Object {
          "type": null,
          "value": "ABSENT",
        },
      },
      Object {
        "descriptor": Object {
          "indent": 0,
          "isEmptyLine": false,
          "isListKey": undefined,
          "lineNo": 524,
          "status": "Passed",
        },
        "expected": Object {
          "type": "int",
          "value": "1",
        },
        "key": Object {
          "type": "key",
          "value": "foo",
        },
        "value": Object {
          "type": "int",
          "value": "1",
        },
      },
    ]
  }
/>
`;
